src/Components/signUp/SignUp.js

```diff
@@ -0,0 +1,245 @@
+import { Fragment, useState, useEffect } from "react";
+import { Navigate } from "react-router-dom";
+
+import { Button, TextField, InputAdornment } from "@mui/material";
+import VisibilityIcon from "@mui/icons-material/Visibility";
+import { VisibilityOff } from "@mui/icons-material";
+
+import Image from "../image/ImageContainer";
+import { styles } from "../../utils";
+import "./styleSignUp.scss";
+
+const SignUp = () => {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [logInAccount, setLogInAccount] = useState(false);
+  const [showPassword, setShowPassword] = useState(false);
+  const [showRepeatPassword, setShowRepeatPassword] = useState(false);
+  const [nameUser, setNameUser] = useState("");
+  const [lastNameUser, setLastNameUser] = useState("");
+  const [emailUser, setEmailUser] = useState("");
+  const [passwordUser, setPasswordUser] = useState("");
+  const [repeatPasswordUser, setRepeatPasswordUser] = useState("");
+  const [messageSignUp, setMessageSignUp] = useState({
+    text: "Enter your detalls below.",
+    messageColor: { color: "rgb(150, 157, 166)" },
+  });
+  const userInformation = JSON.parse(localStorage.getItem("userInformation"));
```

Comment:
lets try to move this code inside `componentDidMount` that way we only execute it once and not every time the component gets render as is happening now